open_audio.py

```diff
@@ -25,35 +27,40 @@
 # Could be done using pydub library also
 #TODO: move to parameters dedicated file
 cut_len_sec = 20
-cut_start_sec = 0 # 60
+cut_start_sec = 60 # 60
 cut_stop_sec = cut_start_sec+cut_len_sec
 left_ch = left_ch[(cut_start_sec*fs):(fs*cut_stop_sec)] 
 t = np.arange(0,cut_len_sec,1/fs)
 
 s_t = left_ch/np.max(left_ch) #normalize to avoid overflow in processing
 O_t = onset_func(s_t)
-x_t = PLL(O_t,fs)
+e_t, x_t = PLL(O_t,fs)
 
-x_t_mod = np.multiply(x_t, np.cos(2*np.pi*440*t) + 0.25*np.cos(2*np.pi*440*3/2*t) + 0.2*np.cos(2*np.pi*440*5/4*t))
-x_t_save = x_t_mod + s_t
-x_t_audio = AudioSegment(
-    ((x_t_save*32767/2).astype('int16')).tobytes(), 
-    frame_rate=fs,
-    sample_width=samples.dtype.itemsize, 
-    channels=1)
-x_t_audio.export("x_t_mod.wav", format="wav")
+# O_t = (1+synth_square(t,2,0.5))/2
+# f_sin = 2
+# O_t = np.sin(2*np.pi*f_sin*t)
+# e_t,x_t = PLL(O_t,fs)
+m_t = metronome_thresholding(x_t)
+
+
+
+# x_t_mod = np.multiply(x_t, np.cos(2*np.pi*440*t) + 0.25*np.cos(2*np.pi*440*3/2*t) + 0.2*np.cos(2*np.pi*440*5/4*t))
+# x_t_save = x_t_mod + s_t
+# x_t_audio = AudioSegment(
+#     ((x_t_save*32767/2).astype('int16')).tobytes(), 
+#     frame_rate=fs,
+#     sample_width=samples.dtype.itemsize, 
+#     channels=1)
+# x_t_audio.export("x_t_mod.wav", format="wav")
 
-x_t_save_2 = x_t + s_t
-# TODO: high pass filter for metronome sound,
-#       fix code (order)
-#       make synthetic inputs as required
-#       
+x_t_save_2 = m_t + s_t
+      
 x_t_audio = AudioSegment(
-    ((x_t*32767).astype('int16')).tobytes(), 
+    ((x_t_save_2*32767/max(abs(x_t_save_2))).astype('int16')).tobytes(), 
```

Comment:
Have the results been better with this normalization?

Reply:
Its not a big issue. I just want to make sure that x has a maximal value of 1 before I scale it to int16 values.
when we add to signal with values in the range [-1,1], we can get values exceeding this range